utils/R/utils.R

```diff
@@ -643,17 +643,18 @@ misc.are.convertible <- function(u1, u2) {
 ##' @export
 ##' @author Istem Fer
 convert.expr <- function(expression) {
-  
-  library(stringi)
-  
+  if (!requireNamespace("stringi", quietly = TRUE)) {
+    logger.error("Package 'stringi' is needed for this function to work. Please install it.")
+  }
+
   # split equation to LHS and RHS
   deri.var <- gsub("=.*$", "", expression) # name of the derived variable
   deri.eqn <- gsub(".*=", "", expression) # derivation eqn
     
   non.match <- gregexpr('[^a-zA-Z_.]', deri.eqn) # match characters that are not "a-zA-Z_."
   split.chars <- unlist(regmatches(deri.eqn, non.match)) # where to split at
   # split the expression to retrieve variable names to be used in read.output
-  variables <- unlist(stri_split_charclass(deri.eqn, paste0("[",noquote(paste0(split.chars, collapse="")),"]"), omit_empty = TRUE))
+  variables <- unlist(stringi::stri_split_charclass(deri.eqn, paste0("[",noquote(paste0(split.chars, collapse="")),"]"), omit_empty = TRUE))
   
```

Comment:
If you're saying that the regular `strsplit` function will return a vector with some empty string elements, can't you just do something like:

``` r
tmp <- strsplit(...)
result <- tmp[tmp != ""]
```

?